app/screens/Canvas.js

```diff
@@ -43,15 +43,20 @@ var Redos = new Array;
 export default Canvas = ({ addDrawing, navigation, route }) => {
   const selectedDrawing = route?.params?.selectedDrawing ? route.params.selectedDrawing : null;
   const selectedFriends = route?.params?.selectedFriends ? route.params.selectedFriends : null;
-  console.log("selectedFriends Canvas:", selectedFriends)
-  console.log("selectedDrawing Canvas:", selectedDrawing)
 
   /*********** Canvas Drawing/Tools States *******/
   var emptyCanvas = new Array
   for (i = 0; i < NUM_ROWS * NUM_COLS; i++) {
     emptyCanvas[i] = eraseColor
   }
-  const [canvasData, setCanvasData] = useState( selectedDrawing != null ? selectedDrawing.pixels : emptyCanvas)
+
+  useEffect(() => {
```

Comment:
@lizardman1028  this is the thing that sets the state multiple times, basically everytime theres a change in selectedDrawing (the thing in the [ ]) this useEffect will run